nightfall-client/src/utils/getCommitmentValues.mjs

```diff
@@ -0,0 +1,114 @@
+import gen from 'general-number';
+import config from 'config';
+import logger from 'common-files/utils/logger.mjs';
+import { randValueLT } from 'common-files/utils/crypto/crypto-random.mjs';
+import Nullifier from '../classes/nullifier.mjs';
+import { findUsableCommitmentsMutex, getSiblingInfo } from '../services/commitment-storage.mjs';
+import Commitment from '../classes/commitment.mjs';
+import { ZkpKeys } from '../services/keys.mjs';
+
+const { GN, generalise } = gen;
+
+const { BN128_GROUP_ORDER } = config;
+
+// eslint-disable-next-line import/prefer-default-export
+export const getCommitmentsValues = async (
+  totalValueToSend,
+  values,
+  recipientZkpPublicKeys,
+  ercAddress,
+  tokenId,
+  rootKey,
+  nonUsableCommitments = [],
+  isFee,
+) => {
+  const { zkpPublicKey, compressedZkpPublicKey, nullifierKey } = new ZkpKeys(rootKey);
+
+  // Generate salts, constrained to be < field size
+  let salts;
+
+  let oldCommitments = [];
+  let newCommitments = [];
+  let leafIndices = [];
+  let localSiblingPaths = [];
+  let blockNumberL2s = [];
+  let roots = [];
+  let nullifiers = [];
+  if (!isFee || totalValueToSend > 0n) {
```

Comment:
This function a bit unclear with all the `lets` and this `if`.

Instead of wrapping all the actual work inside this `if` can we just check the opposite condition and returns the empty array early, i.e. `if(isFee && totalValueToSend <= 0n)`.

Then we won't need all these `lets`.